File: Task06_AgenziaDiViaggi_Angular_CSharp_SQL/Business_Logic/Services/RecensioneService.cs
using Business_Logic.Models;
using Business_Logic.Models.DTO;
using Business_Logic.Repos;

namespace Business_Logic.Services
{
    public class RecensioneService : IService<RecensioneDTO>
    {
        private readonly RecensioneRepo _repo;
        public RecensioneService(RecensioneRepo repo) {
            _repo = repo;
        }

        public RecensioneDTO? Cerca(string varCod)
        {
            throw new NotImplementedException();
        }

        public List<RecensioneDTO> Lista()
        {
            List<RecensioneDTO> recensioniDTO = new List<RecensioneDTO>();
            IEnumerable<Recensione> elencoRecensioni = _repo.GetAll();

            foreach (Recensione rec in elencoRecensioni) {
                if (rec is not null) {
                    RecensioneDTO recDTO = new RecensioneDTO() {
                        codi = rec.Codice,
                        uten = rec.NomeUtente,
                        voto = rec.Voto,
                        text = rec.Commento,
                        data = rec.DataRecensione,
                        pacchettoRIF = rec.PacchettoRIF
                    };
                    recensioniDTO.Add(recDTO);
                }
            }
            return recensioniDTO;
        }
    }
}

File: Task06_AgenziaDiViaggi_Angular_CSharp_SQL/User_Interface/src/app/services/recensione.service.ts
import { Injectable } from '@angular/core';
import { DestinazioneRepo } from '../repos/destinazione.repository';
import { Recensione } from '../models/recensione';
import { RecensioneRepo } from '../repos/recensione.repository';

@Injectable({
  providedIn: 'root'
})
export class RecensioneService {

  constructor(private repo : RecensioneRepo) { }

  ListaRecensioni() : Recensione[]{
    return this.repo.GetAll();
  }

}
